src/coreclr/jit/emit.cpp

```diff
@@ -4820,58 +4820,112 @@ unsigned emitter::getLoopSize(insGroup* igLoopHeader, unsigned maxLoopSize DEBUG
 //                       if currIG has back-edge to dstIG.
 //
 // Notes:
-//    If the current loop encloses a loop that is already marked as align, then remove
-//    the alignment flag present on IG before dstIG.
+//    Despite we align only inner most loop, we might see intersected loops because of control flow
+//    re-arrangement like adding a split edge in LSRA.
+//
+//    If there is an intersection of current loop with last loop that is already marked as align,
+//    then *do not align* one of the loop that completely encloses the other one. Or if they both intersect,
+//    then *do not align* either of them because since the flow is complicated enough that aligning one of them
+//    will not improve the performance.
 //
 void emitter::emitSetLoopBackEdge(BasicBlock* loopTopBlock)
 {
-    insGroup* dstIG = (insGroup*)loopTopBlock->bbEmitCookie;
+    insGroup* dstIG              = (insGroup*)loopTopBlock->bbEmitCookie;
+    bool      noAlignCurrentLoop = false;
```

Comment:
Can you reverse the sense of these, i.e., `alignCurrentLoop` and `alignLastLoop`? It's very confusing sometimes when "no" or "not" is part of a variable name.

Reply:
Sure.